compiler+runtime/src/cpp/jank/runtime/core/seq.cpp

```diff
@@ -707,44 +707,34 @@ namespace jank::runtime
 
   object_ptr merge(object_ptr const m, object_ptr const other)
   {
-    return visit_object(
-      [&](auto const typed_m) -> object_ptr {
-        using T = typename decltype(typed_m)::value_type;
-
-        if constexpr(behavior::associatively_writable<T>)
+    return visit_map_like(
+      [&](auto const typed_other, auto const zero, auto const one) {
+        object_ptr ret{ m };
+        for(auto seq{ typed_other->fresh_seq() }; seq != nullptr; seq = seq->next_in_place())
```

Comment:
We've lost the ability to merge nil with this, I believe.

Reply:
I think previously you'd get an error if `m == nil` (not `associatively_writeable`) and an error if `other == nil` (not `associatively_readable`).

It works to not handle nil because `clojure.core/merge` actually calls `typed_s->conj`, which then calls `runtime::merge` on `this` only if the argument is a map.

Similar for `merge_in_place` via `typed_m->conj_in_place`.

Reply:
`nil` is `associatively_readable` but the previous logic hardcoded the maps instead of checking for the concept.

Reply:
Ah, but it is `associatively_writable`. I will add some checks, but I'm not sure if there's a code path to triggering them.